dashboard/config/initializers/devise.rb

```diff
@@ -283,6 +283,7 @@
 
   config.omniauth :google_oauth2, CDO.dashboard_google_key, CDO.dashboard_google_secret, {
     include_granted_scopes: true,
+    prompt: 'consent',
```

Comment:
Does this mean we're going to prompt every user for access to their Google Classroom information, even if they are only using Google for SSO?

Reply:
nope, only users who have previously given us access to Google Classroom will see the consent page for Google Classroom

Reply:
On each login? I'm getting this second prompt every time I log out and log back in:

![image](https://user-images.githubusercontent.com/413693/55109418-11c44200-5093-11e9-8862-5aec8d8b9ed0.png)

Reply:
yes, that will happen on each login for teachers who have previously given consent for Google Classroom access. happy to talk through alternative approaches!

Reply:
Would it be possible to only prompt if the user doesn't have a refresh token already?

Reply:
that was my original intent/hope, but there are a few impediments with that:
1. the "consent" step comes before we know anything about the user, so we can't check if we have a refresh token beforehand
2. we can't assume that a refresh token is valid by its presence. refresh tokens are invalidated when a user changes their password or revokes access to code.org then comes back (the latter being very unlikely and not an actual concern)

i think there are a few good alternative options, but they require more time/work:
1. separate the sign in and sign up paths. currently, clicking "continue with google" takes you to a generic route that _decides_ if you're signing in or signing up. if these were separated somehow, and we _knew_ that the user was trying to sign up, we could request consent, whereas if we knew the user was signing in, we would skip the consent step.
2. integrate this consent step into rostered section functionality. when a teacher is signed in and trying to create a google classroom section or sync an existing section, we could automatically detect invalid tokens or a missing refresh token and re-prompt the user for their consent